From 51a0ac0ff3a726f88af9e8cd472779b33117d304 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Sun, 3 Feb 2019 14:34:42 +0100
Subject: [PATCH 31/32] Prevent static build to prefer dynamic libraries

The static build should not attempt to link against dynamic libraries
in order to get a fully statically linked binary in the end.
---
 mkspecs/features/qt_configure.prf | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/mkspecs/features/qt_configure.prf b/mkspecs/features/qt_configure.prf
index 81110e9f6d..61f7cf27d1 100644
--- a/mkspecs/features/qt_configure.prf
+++ b/mkspecs/features/qt_configure.prf
@@ -577,12 +577,12 @@ defineTest(qtConfResolveLibs) {
                 # Under Windows, we look only for static libraries, as even for DLLs
                 # one actually links against a static import library.
                 mingw {
-                    lcan += \
-                        # MinGW supports UNIX-style library naming in addition to
-                        # the MSVC style.
-                        lib$${lib}.dll.a lib$${lib}.a \
-                        # Fun fact: prefix-less libraries are also supported.
-                        $${lib}.dll.a $${lib}.a
+                    # prefer (import) libraries for dynamic linking unless we're doing a
+                    # static build where we solely rely on presence of static libraries
+                    shared {
+                        lcan += $${QMAKE_PREFIX_STATICLIB}$${lib}.$${QMAKE_EXTENSION_IMPORTLIB} $${lib}.$${QMAKE_EXTENSION_IMPORTLIB}
+                    }
+                    lcan += $${QMAKE_PREFIX_STATICLIB}$${lib}.$${QMAKE_EXTENSION_STATICLIB} $${lib}.$${QMAKE_EXTENSION_STATICLIB}
                 }
                 lcan += $${lib}.lib
             }
-- 
2.24.0

